internal/controller/dnsrecord_healthchecks.go

```diff
@@ -94,6 +95,69 @@ func (r *DNSRecordReconciler) ensureProbe(ctx context.Context, generated *v1alph
 	return nil
 }
 
+// removeUnhealthyEndpoints fetches all probes associated with this record and uses the following criteria while removing endpoints:
+//   - If the Leaf Address has no health check CR - it is healthy
+//   - If the health check CR has insufficient failures - it is healthy
+//   - If the health check CR is deleting - it is healthy
+//   - If the health check is a CNAME and any IP is healthy - the CNAME is healthy
```

Comment:
Is it also the case that if it is the last endpoint it healthy?

Reply:
Nope, if this would remove all the enpoints we will do nothing. I'll add more comments

Reply:
Discussed. We wont remove records only will remove if multi-value record